components/chainhook-cli/src/config/file.rs

```diff
@@ -43,11 +45,36 @@ pub struct LimitsConfigFile {
 
 #[derive(Deserialize, Debug, Clone)]
 pub struct NetworkConfigFile {
-    pub mode: String,
+    pub mode: NetworkConfigMode,
     pub bitcoind_rpc_url: String,
     pub bitcoind_rpc_username: String,
     pub bitcoind_rpc_password: String,
     pub bitcoind_zmq_url: Option<String>,
     pub stacks_node_rpc_url: Option<String>,
     pub stacks_events_ingestion_port: Option<u16>,
 }
+
+#[derive(Deserialize, Debug, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum NetworkConfigMode {
```

Comment:
Why not re-using the one of the `Network` structs available in the chainhook-types module?

Reply:
Mainly because of the `from_bitcoin_network` function that I needed. We need to be take `BitcoinNetwork::Regtest` (which could be provided to the generator) and convert it to `Devnet`. Perhaps I could add that function to the `StacksNetwork` struct?